public/app/plugins/datasource/prometheus/components/PromQueryField.tsx

```diff
@@ -163,9 +163,10 @@ class PromQueryField extends React.PureComponent<PromQueryFieldProps, PromQueryF
     const {
       data,
       datasource: { languageProvider },
+      range,
     } = this.props;
 
-    if (languageProvider !== prevProps.datasource.languageProvider) {
+    if (languageProvider !== prevProps.datasource.languageProvider || range !== prevProps.range) {
```

Comment:
```suggestion
    if (languageProvider !== prevProps.datasource.languageProvider || !_.isEqual(range, prevProps.range)) {
```

As we are comparing objects, maybe we could use deep comparison to make sure that the values have changed.